### ISSUE TITLE
tui grid datasource and tui pagination

### ISSUE THREAD

**Issue by kimsiwon** [nhn/tui.grid#918]
안녕하세요.. 제공해주시는 tui grid로 컨버전 작업중 막히는 부분이 있습니다..
vue.js로 개발 하였습니다. 버전은 tui grid- 4.9 , tui pagination - 3.5 입니다.
spa로 개발중이고 @/api/manager/menu 라는 경로에 export function getEtcORG() {} 형태의 resful api가 있습니다. 해당 resful api는 url, method, params, header meta데이터등을 저희가 사전에 만들어둔 유틸 컴포넌트로 태워 해당 url로 rest요청을 합니다..

tui grid 생성을할때 data 영역에 api: { readData: { url: '@/api/manager/menu', method: 'GET' } } 이라고 넣으면 되는게 아닌가요? status code는 200ok가 떨어졌는데 응답전문에서는 'we're sorry but preset doesn't work properly without javascript enalbled' 라는 에러가나네요
콘솔창에서 는 unexpected token in json at position 0 라는 에러가 뜹니다
저희가 백단에서 데이터를 가져올때 json형태로 클라이언트로 보내는데 문제가 되나요?

그리고 올려주신 이전 글들을 참고하며 mounted() 영역에 this.$refs.tuiGrid.invoke('setPerPage', 1)도 작성하였는데 동작하지않습니다..

그리드 옵션에는 pageOptions: { perPage: 5 } 라는 옵션을 주었습니다..

구글링을 하루종일해도 자료가 부족함이 많이 느껴집니다. 오픈소스를 제공해주셔서 감사하기는 하나 nhn.github.io 사이트의 api와 예제들도 어떤 옵션들과 메서드들이 있는지 알수 있지만 실제로 개발할때 도움이 되지 않습니다..

vue.js로 datasource와 pagination 구현되어있는 예제가 있으시면 도와주세요..
전체 패키지를 주시면 더 감사합니다.. 구성이 이해가 잘 안되서요..
backend에 perPage등의 변수를 받아서 처리하여 반환해주는것인지 backend에서 데이터를 전체 다 받아와서 perpage등의 옵션으로 분기하여 페이징처리하는것인지 알고싶습니다

도와주세요 상당히 곤혹을 치르고있습니다 부탁드립니다 살려주세요

**Summary**
tui grid datasource and tui pagination

**Screenshots**
If applicable, add screenshots to help explain your question.

**Version**
tui grid - 4.9 , tui pagination - 3.5

**Additional context**

**Comment by js87zz** [comment]
@kimsiwon 
The `invoke` function is only used in our [vue wrapper](https://github.com/nhn/tui.grid/tree/master/packages/toast-ui.vue-grid) and cannot be used in `tui-grid`. I think it's better that you install the **vue wrapper** and use it.
```js
this.$refs.tuiGrid.invoke('setPerPage', 1)
```

If you pass the `url` option like below, it's wrong syntax regardless of our grid. You have to pass the full request path.
```js
const url = getEtcORG();
data: {
  api: {
    // You should have pass the full request path(not using alias)
    readData: { url: '@/api/manager/menu', method: 'GET' }

    readData: { url, method: 'GET' }
  }
}
```

Refer this [tutorial](https://github.com/nhn/tui.grid/blob/master/packages/toast-ui.grid/docs/ko/data-source.md). it will be helpful.

If you have another trouble shooting, you give us to more detail source code(grid configuration). 

Lastly, Please make sure you have filled out issue respecting our form **in English** and given us as much context as possible.

**Comment by kimsiwon** [comment]
@js87zz

thank you for your answer.. but
why fill out issue respecting our form in English?
where is Dong woo Kim teacher?

i'm sorry for can't give to you with detail source code
because my company is very forced security..

you can't understand my korean question?

clearly, i told you already try your way @api/manager/menu 

my project exist getEtcOrg() function in menu.js file . then in data : {} no express function name.
how that's example know getEtcOrg() function..

because my env is hard to give my source code, im sorry but i want you to grive me a vue.js + datasource + pagination please..

and my last question is if I am receiving a variable such as perPage at the backend and processing it, or if I am receiving all the data from the backend and branching it with options such as perpage.

in summary my situation , i used tui grid + pagination , because using datasource my restful api is in getEtcOrg() function in @api/manager/menu.js but i used readData: { url: @/api/manager/menu, method: get} get error  'we're sorry but preset doesn't work properly without javascript enalbled' on response specialty..

please read my reply cafuuly and answer.. please..

**Comment by js87zz** [comment]
@kimsiwon 
I understood your korean question clearly. It's sorry that If my answer has not been helpful enough. You also need to change your aggressive attitude for asking the question. This isn't a battlefield. I read your question again and comment below. 

1. why fill out issue respecting our form in English?
- It's the policy we decided to share the issue with other foreign people. They cannot be understand the korean issue.

2. RESTful URI
- I commented above about your weird syntax. Relative import path cannot be RESTful URI.(`@api/manager/menu` is not request url, it's just your module relative path)
- If you define the RESTful url, define the function in `url` or the result of your `getEtcOrg` function.
```js
import { getEtcOrg } from '@api/manager/menu';

// it's possible to call `getEtcOrg` function directly, you can use below
const url = getEtcOrg(someParam);

data: {
  api: {
    readData: { url, method: 'GET' }
  }
}

// or define the function url
data: {
  api: {
    // example
    readData: { url: () => getEtcOrg(someParam), method: 'GET' }
  }
}

// if 
```

3. paging
- With `dataSource`, the data should be paginated in backend by `perPage` and grid will render the only paginated data(not all data)
